tests/test_type_aliases.py

```diff
@@ -0,0 +1,27 @@
+from typing import Dict, List
+
+import pytest
+
+from stable_baselines3.common.type_aliases import check_cast, non_null
+
+
+def test_non_null():
+    for a in (1, "a", [2]):
+        assert non_null(a) == a
+
+    with pytest.raises(ValueError):
+        non_null(None)
+
+
+def test_check_cast():
+    assert check_cast(dict, {}) == {}
+    assert check_cast(dict[str, int], {}) == {}
+    assert check_cast(Dict[str, int], {}) == {}
+
+    with pytest.raises(TypeError):
+        check_cast(list[int], {})
+        check_cast(List[int], {})
+
+    # NOTE: check_cast does not check the template arguments, only the main class.
```

Comment:
It might be good to include this caveat in a docstring for `check_cast` as well.

I'm also not totally sure how I feel about adding a unit test ensuring that a bug (check_cast doesn't check the template arguments) continues to fail.

Reply:
> ensuring that a bug (check_cast doesn't check the template arguments) continues to fail

I think tests should give an accurate understanding of how the function works, so I've left this assert in but added a note that triggering it is *good*. ffccebb2

Reply:
Works for me, still not sure how I feel about it but definitely don't feel strongly enough to argue.